parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,272 @@
+            numStatementsParsed += 1;
+        }
+        else {
+            writer.write(statement + " ----> " + errorType);
+            writer.write("\n\n");
+        }
+    }
+
+    private class SqlLogicTestFileVisitor
+            implements FileVisitor<Path>
+    {
+        private void processStatement(BufferedReader reader)
+                throws IOException
+        {
+            StringBuilder statement = new StringBuilder();
+            while (true) {
+                final String line = reader.readLine();
+                if (line == null || isLineAfterStatement(line)) {
+                    break;
+                }
+                statement.append(line);
+            }
+            if (!statement.substring(statement.length() - 1).equals(";")) {
+                statement.append(";"); // NOTE: This semi-colon is necessary for parsing.
+            }
+            numStatementsFound += 1;
+            validateParsingOfSqlStatement(statement.toString());
+        }
+
+        private boolean isLineBeforeStatement(String line)
+        {
+            return line.startsWith("statement ok") || line.startsWith("query");
+        }
+
+        private boolean statementIsAcceptableToCurrRdms()
+        {
+            if (rdms == AcceptedRdms.ALL) {
+                return skipIfSet.isEmpty() && onlyIf == null;
+            }
+
+            if (onlyIf != null && onlyIf != rdms) {
+                return false;
+            }
+            return !skipIfSet.contains(rdms);
+        }
+
+        private boolean isLineAfterStatement(String line)
+        {
+            return line.equals("----") || line.equals("");
+        }
+
+        private boolean processConditionalRecord(String line)
+        {
+            if (line.startsWith("onlyif")) {
+                String onlyIfRdms = line.split(" ")[1];
+                for (AcceptedRdms acceptedRdms : AcceptedRdms.values()) {
+                    if (acceptedRdms.toString().equalsIgnoreCase(onlyIfRdms)) {
+                        onlyIf = acceptedRdms;
+                    }
+                }
+                return true;
+            }
+
+            if (line.startsWith("skipif")) {
+                String skipIfRdms = line.split(" ")[1];
+                for (AcceptedRdms acceptedRdms : AcceptedRdms.values()) {
+                    if (acceptedRdms.toString().equalsIgnoreCase(skipIfRdms)) {
+                        skipIfSet.add(acceptedRdms);
+                    }
+                }
+                return true;
+            }
+            return false;
+        }
+
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
+        {
+            return CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                throws IOException
+        {
+            if (Files.isHidden(file) || numFilesVisited > Integer.parseInt(propertiesMap.get("max.files.to.visit"))) {
```

Comment:
Probably SqlLogicTest shouldn't need to know about the string literal "max.files.to.visit" - it would be better if this was passed into the constructor or was parsed out there.

Reply:
This literal and the logic around it is in the Config class now.

Done.

Reply:
Factored out in new config design.

Done